### PR TITLE
EOS-18545: Sensor classes should read common config from one place no…

### PR DESCRIPTION
**Pull request by gargpalak048**
…t in individual class.

## Problem Statement
<pre>
  <code>
    Story Ref (if any):
    Every Sensor or Msg Handler class retrieves the common alert config individually which adds a lot of code repetition, a 
    utility class/function will make it clean.
  </code>
</pre>
## Problem Description
<pre>
  <code>
    Please add Problem decription here...
  </code>
</pre>
## Solution
<pre>
  <code>
    Move common code in conf_utils file and every sensor should read config from there.
  </code>
</pre>
## Sanity testing on RPM done
<pre>
  <code>
    - [ ] Yes
    - [ ] No
  </code>
</pre>
## Unit/Manual Testing Description
<pre>
  <code>
    Please describe the tests that you ran to verify your changes.
  </code>
</pre>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: low-level/framework/utils/conf_utils.py [hunk 1]
```diff
@@ -148,3 +154,9 @@
 
 SRVNODE = Conf.get(GLOBAL_CONF, f'{CLUSTER}>{SERVER_NODES}')[MACHINE_ID]
 ENCLOSURE = Conf.get(GLOBAL_CONF, f"{CLUSTER}>{SRVNODE}>{STORAGE}>{ENCLOSURE_ID}")
+
+# Get SYSTEM INFORMATION specific config to use in other files.
+GLOBAL_SITE_ID = Conf.get(GLOBAL_CONF, f"{CLUSTER}>{SRVNODE}>{SITE_ID}",'DC01')
```
Comment (cdeshmukh) [inline low-level/framework/utils/conf_utils.py:159]: I dont think GLOBAL needs to be prepended, as there is no LOCAL version of it. Just keep SITE_ID & alike for others

File: low-level/framework/utils/mon_utils.py [hunk 2]
```diff
@@ -28,3 +30,8 @@ def get_alert_id(epoch_time):
     salt = str(uuid.uuid4().hex)
     alert_id = epoch_time + salt
     return alert_id
+
+def get_epoch_time_from_date_and_time(date, time):
```
Comment (cdeshmukh) [inline low-level/framework/utils/mon_utils.py:34]: keep name short get_epoch() is enough. Based on arguments, people can infer further, or later we can add/alter arguments too
Also this is generic utility function, can stay in uility.py

File: low-level/framework/utils/utility.py [hunk 3]
```diff
@@ -79,3 +81,42 @@ def get_os(self):
...
+    def create_file(self, path):
+        """Create a file at specified path"""
+        dir = path[:path.rindex("/")]
+        if not os.path.exists(dir):
+            os.makedirs(dir)
+        if not os.path.exists(path):
+            file = open(path, "w+")
+            file.close()
+
+    def read_data(self, file):
```
Comment (cdeshmukh) [inline low-level/framework/utils/utility.py:93]: read_raw_filedata()

File: low-level/framework/utils/utility.py [hunk 4]
```diff
@@ -79,3 +81,42 @@ def get_os(self):
...
+            from subprocess import run, PIPE
+            from re import findall
+
+            IP_CMD = "ip -f inet addr show scope global up | grep inet"
+            IP_REGEX = b'\\b(\\d{1,3}(?:\\.\d{1,3}){3})/\d{1,2}\\b'
+
+            ip_out = run(IP_CMD, stdout=PIPE, shell=True, check=True)
+            ip_list = re.findall(IP_REGEX, ip_out.stdout)
+            if ip_list:
+                return ip_list[0]
```
Comment (cdeshmukh) [inline low-level/framework/utils/utility.py:116]: can we logger.debug() hostname before any 'return'

File: low-level/framework/utils/conf_utils.py (hunk 1 above)
Comment (gargpalak048) [reply low-level/framework/utils/conf_utils.py:159]: <SITE_ID> var name is already used above.
#L122 SITE_ID="site_id"
Still changes required?

File: low-level/framework/utils/utility.py (hunk 4 above)
Comment (gargpalak048) [reply low-level/framework/utils/utility.py:125]: Done.

File: low-level/framework/utils/utility.py (hunk 3 above)
Comment (gargpalak048) [reply low-level/framework/utils/utility.py]: Done.

File: low-level/framework/utils/mon_utils.py (hunk 2 above)
Comment (gargpalak048) [reply low-level/framework/utils/mon_utils.py]: Done the changes.
Moved the get_epoch() function to Utility.py
Also, Can't we move get_alert_id() func to Utiliy.py too?
why to keep separate mon_utils.py file for that?

Comment (cdeshmukh) [reply low-level/framework/utils/mon_utils.py]: Monitoring specific ulity functions maintained in mon_utils

File: low-level/framework/utils/conf_utils.py (hunk 1 above)
Comment (cdeshmukh) [reply low-level/framework/utils/conf_utils.py:159]: Why this GLOBAL_ seen still ?

**Comment by gargpalak048** [comment]
Old PR

**Pull request closed without merging**